Add docs for github sync environment secrets (#28175)
hashicorp/vault: website/content/api-docs/system/secrets-sync.mdx

### Parameters

> - `name` `(string: <required>)` - Specifies a custom name for the GitHub application that is used when configuring
> the destination. Although this can be any name, we recommend using the same app name configured in GitHub.

Reviewer: ```suggestion
- `name` `(string: <required>)` - Specifies a custom name for the GitHub
  application when configuring the sync destination. We recommend using the same 
  app name configured in GitHub.
```
Style correction: write in active voice

Also, why is using the same name considered best practice? Is it just so you can
tell at a glance which app the configuration relates to or is there another
reason?

Author: Yes, it's exactly that.
